src/software/ai/navigator/obstacle/new_obstacle/polygon_obstacle.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "software/ai/navigator/obstacle/new_obstacle/obstacle.h"
+#include "software/new_geom/polygon.h"
+
+/**
+ * An Obstacle represented by a Polygon
+ */
+class PolygonObstacle : public Obstacle
+{
+   public:
+    PolygonObstacle() = delete;
+
+    /**
+     * Construct a polygon obstacle by with a Polygon
+     * @param polygon Polygon to make obstacle with
+     */
+    explicit PolygonObstacle(const Polygon& polygon);
+
+    bool contains(const Point& p) const override;
+    double minDistance(const Point& p) const override;
+    bool intersects(const Segment& segment) const override;
+    std::string toString(void) const override;
+
+    /**
+     * Gets the underlying polygon
+     *
+     * @return polygon
+     */
+    const Polygon getPolygon(void) const;
+
+   private:
+    Polygon polygon_;
```

Comment:
Similar comment here, why not have `PolygonObstacle` extend `Obstacle` and `Polygon`?